src/kaleidoscope/Kaleidoscope.h

```diff
@@ -132,6 +132,28 @@ class Kaleidoscope_ {
     return millis_at_cycle_start_;
   }
 
+  /** Determines if a timer has expired.
+   *
+   * This method should be used whenever checking to see if a timeout has been
+   * reached. It correctly computes timeout values even when integer overflow
+   * occurs, given a start time and a timeout. It takes two parameters:
+   *
+   * - start_time: A timestamp when the timer started, which should be set by
+   *      calling `Kaleidoscope.millisAtCycleStart()`. It can be any integer
+   *      type.
+   *
+   * - ttl: The timeout value or interval to check (ttl = "time to live"). The
+   *      timer expires (and `hasTimeExpired()` returns true) when the time that
+   *      has elapsed since `start_time` exceeds this value. It must be and
+   *      integer type that is no bigger than the type of `start_time`.
+   */
+  template <typename _Timestamp = uint16_t, typename _Timeout = uint16_t>
```

Comment:
That's about the first time that I see default values for a template function.

Default values for template parameters make sense when defining template classes. The only place where I can see them to be useful with template functions is when taking a pointer to the respective function. There you could write 
```cpp
auto fncPtr = &hasTimeExpired<>;
```
Is that intented? Or what is the reason for the default parameters here?

Reply:
You give me more credit than I deserve in this case; my inexperience with C++ templates is showing. Now that you've pointed it out, it does seem pointless to use default values, though I should add something more suggesting that it's best to use the smallest size that will work (and that, because integer overflow is handled properly, it will often work to use relatively small sizes).

Reply:
See #641 for an update.